### ISSUE TITLE
Add common regularizations to PyTorch.

### ISSUE THREAD

**Issue by veritas9872** [pytorch/pytorch#68514]
## 🚀 Feature
I propose including L1/L2 regularization and perhaps stochastic depth to the PyTorch library.

## Motivation

PyTorch does not currently support L1 regularization natively despite its once being a very popular method.
Indeed, the decline of L1 may in part be due to the lack of a native PyTorch implementation.
Also, the L2 regularization offered by PyTorch is specified in the optimizer and cannot change in value.
Finally, stochastic depth, despite being a standard technique in training residual neural networks, must be implemented by users, making it impossible to use pre-trained models directly.

## Pitch

I propose adding the L1 and L2 regularizations as backward hooks, hopefully with the option of changing the regularization terms during training.
Stochastic depth may be included as a layer. I do not know how it could be implemented as a hook.

These proposals have already been implemented by TorchLayers, though L1 and L2 are layers instead of pure hooks.
https://github.com/szymonmaszke/torchlayers/blob/master/torchlayers/regularization.py

 
## Alternatives

Perhaps the optimizer could automatically perform regularization of either L1 or L2 or a combination of the two.
The fact that the regularization values cannot be changed is very frustrating because the loss generally declines while the weights generally increase during training.

## Additional context

L1 and L2 regularization have declined in popularity in recent years despite their initial popularity. This may be due to the inconvenience of re-implementing them in PyTorch, whereas they are available natively in Keras.
Also, despite the widespread popularity of learning rate scheduling, there is very little research into regularization scheduling, even though they both operate on the ease of learning.
Stochastic depth is also made more difficult than necessary.
I believe that adding these simple regularizations will be immensely helpful to the research community at large.
